src/OrchardCore.Modules/OrchardCore.OpenId/Services/OpenIdServerService.cs

```diff
@@ -474,22 +553,28 @@ private static DirectoryInfo GetSigningCertificateDirectory(ShellOptions options
 
         private async Task<ImmutableArray<(string path, X509Certificate2 certificate)>> GetCertificatesAsync(DirectoryInfo directory)
         {
-            if (!directory.Exists)
-            {
-                return ImmutableArray.Create<(string, X509Certificate2)>();
-            }
-
             var certificates = ImmutableArray.CreateBuilder<(string, X509Certificate2)>();
 
-            foreach (var file in directory.EnumerateFiles("*.pfx", SearchOption.TopDirectoryOnly))
+            List<FileInfo> files = new List<FileInfo>();
+            try { files.AddRange(directory.EnumerateFiles("*.pfx", SearchOption.TopDirectoryOnly)); } catch (DirectoryNotFoundException) {};
+            files.AddRange(_linuxThumbprintsOfAvailableCertificates.Select(thumbprint => new FileInfo(Path.Combine(directory.FullName, thumbprint + ".pfx"))));
+            files.AddRange(_linuxThumbprintsOfAvailableCertificates.Select(thumbprint => new FileInfo(Path.Combine(directory.FullName, thumbprint + ".p12"))));
+            files.AddRange(_linuxThumbprintsOfAvailableCertificates.Select(thumbprint => new FileInfo(Path.Combine(directory.FullName, thumbprint + ".der"))));
+            
+            foreach (var file in files)
             {
                 try
                 {
                     // Only add the certificate if it's still valid.
                     var certificate = await GetCertificateAsync(file.FullName);
-                    if (certificate.NotBefore <= DateTime.Now && certificate.NotAfter > DateTime.Now)
+                    if (certificate != null)
                     {
-                        certificates.Add((file.FullName, certificate));
+                        _logger.LogError("certificate not null: NotBefore:" + certificate.NotBefore + " NotAfter:" + certificate.NotAfter);
+                        if (certificate.NotBefore <= DateTime.Now && certificate.NotAfter > DateTime.Now)
```

Comment:
Cert for signing+encryption dont have to be "valid". Key rotation is a different thing that should be solved here.